### ISSUE TITLE
Weird behaviour with numbered lists and list-item-bullet-indent

### ISSUE THREAD

**Issue by angrykoala** [remarkjs/remark-lint#206]
I have remark-lint and remark cli with the following .remarkrc

```json
{
    "plugins": [
        "remark-preset-lint-recommended",
        ["remark-lint-list-item-indent", true]
    ]
}
```

The bullet lists are working fine for the most part, with the following example passing the linting:

```markdown
* foo
  * bar
  * bar 2
* foo 2
```

However, when mixin ordered lists and bullet lists:

```markdown
1. foo
  * bar
  * bar 2
2. foo 2
```

The lint shows the error `Incorrect indentation before bullet: remove 2 spaces (remark-lint:list-item-bullet-indent)` I'm not sure why this error happen and if there is a way to make it recognize ordered and unordered lists mixed together for indentation

**Comment by ChristianMurphy** [comment]
@angrykoala for nested lists, the inner list's marker item marker needs to be indented as much or more than content of the parent list item. https://spec.commonmark.org/0.28/#lists

In the second example:

```markdown
1. foo
  * bar
  ^ this marker needs one more space to match `foo`
```

the following content is valid:

```markdown
*   foo
    *   bar
    *   bar 2
*   foo 2


1.  foo
    *   bar
    *   bar 2
2.  foo 2
```
